whois-api/src/test/java/net/ripe/db/whois/api/rest/OverrideWithNonTrustedIpTestIntegration.java

```diff
@@ -0,0 +1,211 @@
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.common.rpsl.RpslObjectBuilder;
+import org.glassfish.jersey.uri.UriComponent;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import java.time.LocalDateTime;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.fail;
+
+// FIXME: make this into a suite that runs twice: once with XML, once with JSON
+@Category(IntegrationTest.class)
+public class OverrideWithNonTrustedIpTestIntegration extends AbstractIntegrationTest {
+
+    private static final RpslObject PAULETH_PALTHEN = RpslObject.parse("" +
+            "person:    Pauleth Palthen\n" +
+            "address:   Singel 258\n" +
+            "phone:     +31-1234567890\n" +
+            "e-mail:    noreply@ripe.net\n" +
+            "mnt-by:    OWNER-MNT\n" +
+            "nic-hdl:   PP1-TEST\n" +
+            "remarks:   remark\n" +
+            "source:    TEST\n");
+
+    private static final RpslObject OWNER_MNT = RpslObject.parse("" +
+            "mntner:      OWNER-MNT\n" +
+            "descr:       Owner Maintainer\n" +
+            "admin-c:     TP1-TEST\n" +
+            "upd-to:      noreply@ripe.net\n" +
+            "auth:        MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+            "auth:        SSO person@net.net\n" +
+            "mnt-by:      OWNER-MNT\n" +
+            "source:      TEST");
+
+    private static final RpslObject TEST_PERSON = RpslObject.parse("" +
+            "person:    Test Person\n" +
+            "address:   Singel 258\n" +
+            "phone:     +31 6 12345678\n" +
+            "nic-hdl:   TP1-TEST\n" +
+            "mnt-by:    OWNER-MNT\n" +
+            "source:    TEST\n");
+
+
+    @Autowired private WhoisObjectMapper whoisObjectMapper;
+    @Autowired private TestDateTimeProvider testDateTimeProvider;
+
+    @Before
+    public void setup() {
+        databaseHelper.addObject("person: Test Person\nnic-hdl: TP1-TEST");
+
+        databaseHelper.insertUser(User.createWithPlainTextPassword("db_e2e_1", "zoh", ObjectType.PERSON));
+        databaseHelper.insertUser(User.createWithPlainTextPassword("person", "zoh", ObjectType.PERSON));
+
+        databaseHelper.addObject(OWNER_MNT);
+        databaseHelper.updateObject(TEST_PERSON);
+        testDateTimeProvider.setTime(LocalDateTime.parse("2001-02-04T17:00:00"));
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
```

Comment:
Reset the ipranges.trusted value to whatever it was before in a @AfterClass method to prevent test failures in other tests.

Reply:
Every test case reloads the test properties file  and it is defined there. So should not be an issue
But still I will clear the property in afterclass